server/lib/statistics/getSettingsStatistics.ts

```diff
@@ -0,0 +1,160 @@
+			apps: {
+				frameworkDevMode,
+				frameworkEnable,
+			},
+			nps: {
+				surveyEnabled,
+			},
+			update: {
+				updateChecker,
+			},
+		},
+		liveStreamAndBroadcasting: {
+			liveStream,
+			broadcasting,
+		},
+		message: {
+			allowEditing,
+			allowDeleting,
+			allowUnrecognizedSlashCommand,
+			allowBadWordsFilter,
+			readReceiptEnabled,
+			readReceiptStoreUsers,
+		},
+		otr: {
+			otrEnable,
+		},
+		push: {
+			pushEnable,
+		},
+		search: {
+			defaultProvider: {
+				globalSearchEnabled,
+			},
+		},
+		threads: {
+			threadsEnabled,
+		},
+		videoConference: {
+			bigBlueButton,
+			jitsiEnabled,
+		},
+		webRTC: {
+			webRTCEnableChannel,
+			webRTCEnablePrivate,
+			webRTCEnableDirect,
+		},
+	};
+	return statisticObject;
+};
+export const getSettingsStatistics = async (): Promise<ISettingStatisticsObject> => {
+	try {
+		const settingsBase = [
+			{ key: 'Accounts_TwoFactorAuthentication_Enabled', alias: 'account2fa' },
+			{ key: 'E2E_Enable', alias: 'e2e' },
+			{ key: 'E2E_Enabled_Default_DirectRooms', alias: 'e2eDefaultDirectRoom' },
+			{ key: 'E2E_Enabled_Default_PrivateRooms', alias: 'e2eDefaultPrivateRoom' },
+			{ key: 'SMTP_Host', alias: 'smtpHost' },
+			{ key: 'SMTP_Port', alias: 'smtpPort' },
+			{ key: 'From_Email', alias: 'fromEmail' },
+			{ key: 'Apps_Framework_Development_Mode', alias: 'frameworkDevMode' },
+			{ key: 'Apps_Framework_enabled', alias: 'frameworkEnable' },
+			{ key: 'NPS_survey_enabled', alias: 'surveyEnabled' },
+			{ key: 'Update_EnableChecker', alias: 'updateChecker' },
+			{ key: 'Livestream_enabled', alias: 'liveStream' },
+			{ key: 'Broadcasting_enabled', alias: 'broadcasting' },
+			{ key: 'Message_AllowEditing', alias: 'allowEditing' },
+			{ key: 'Message_AllowDeleting', alias: 'allowDeleting' },
+			{ key: 'Message_AllowUnrecognizedSlashCommand', alias: 'allowUnrecognizedSlashCommand' },
+			{ key: 'Message_AllowBadWordsFilter', alias: 'allowBadWordsFilter' },
+			{ key: 'Message_Read_Receipt_Enabled', alias: 'readReceiptEnabled' },
+			{ key: 'Message_Read_Receipt_Store_Users', alias: 'readReceiptStoreUsers' },
+			{ key: 'Search.defaultProvider.GlobalSearchEnabled', alias: 'globalSearchEnabled' },
+			{ key: 'OTR_Enable', alias: 'otrEnable' },
+			{ key: 'Push_enable', alias: 'pushEnable' },
+			{ key: 'Threads_enabled', alias: 'threadsEnabled' },
+			{ key: 'bigbluebutton_Enabled', alias: 'bigBlueButton' },
+			{ key: 'Jitsi_Enabled', alias: 'jitsiEnabled' },
+			{ key: 'WebRTC_Enable_Channel', alias: 'webRTCEnableChannel' },
+			{ key: 'WebRTC_Enable_Private', alias: 'webRTCEnablePrivate' },
+			{ key: 'WebRTC_Enable_Direct', alias: 'webRTCEnableDirect' },
+			{ key: 'Canned_Responses_Enable', alias: 'cannedResponses' },
+		];
+
+		// Mapping only _id values
+		const settingsIDs = settingsBase.map((el) => el.key);
+
+		const settingsArray = await Settings.findByIds(settingsIDs).toArray();
+		const settingsStatistics = settingsArray
+			.map((el): ISettingStatistics => {
```

Comment:
I believe the `Cursor` object returned by `Settings.findByIds()` has a `.map()` method:

```suggestion
		// no intermediary `settingsArray`
		const settingsStatistics = await Settings.findByIds(settingsIDs)
			.map((el): ISettingStatistics => {
```

But the result of it may be `Promise<ISettingStatistics[]>`, so you may need an `await` before applying `.filter` and `.reduce` methods.

Reply:
Unfortunately the `Cursor` return only `ISetting` and to get `ISetting[]` is necessary `.toArray()`

Reply:
`Cursor<ISetting>` is almost like a stream of `ISetting` objects, where `.toArray()` works like a "collector" or "aggregator". You could chain `.map()` and `.filter()` generating new `Cursor<T>` instances before invoking `.toArray()`. Take a look at my commit d077c8f.